test/DacpacTool.Tests/PackageBuilderTests.cs

```diff
@@ -605,17 +610,69 @@ public void GenerateCreateScript_IncludeCompositeObjects(bool includeCompositeOb
             packageBuilder.AddReference(secondReference);
             packageBuilder.ValidateModel();
 
+            var buildOptions = new BuildOptions();
+            buildOptions.Output = tempFile;
+            buildOptions.TargetDatabaseName = packageName;
+
+            if (includeCompositeObjects.HasValue)
+            {
+                buildOptions.CreateScriptProperty = new[] { $"IncludeCompositeObjects={includeCompositeObjects}" };
+            }
+
+            // Act
+            packageBuilder.SaveToDisk(tempFile);
+            packageBuilder.GenerateCreateScript(buildOptions);
+
+            // Assert
+            var scriptFilePath = Path.Combine(tempFile.DirectoryName, $"{packageName}_Create.sql");
+            File.Exists(scriptFilePath).ShouldBeTrue();
+
+            var shouldContainReferencedTables = includeCompositeObjects == true;
+
+            var scriptContent = File.ReadAllText(scriptFilePath);
+            scriptContent.Contains("CREATE TABLE [dbo].[MyFirstTable]").ShouldBe(shouldContainReferencedTables);
+            scriptContent.Contains("CREATE TABLE [dbo].[MySecondTable]").ShouldBe(shouldContainReferencedTables);
+
+            // Cleanup
+            tempFile.Delete();
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        [DataRow(null)]
+        public void GenerateCreateScript_CreateNewDatabase(bool? createNewDatabase)
```

Comment:
It feels a bit weird to have these seperate test cases for what is essentially the same logic, which is making sure we can pass along deployment properties.

Reply:
I can remove these tests but I think we should be 100% sure the deployment options are not modified inside the `GenerateCreateScript` method.

I can add a test where I first generate the script using `DacServices.GenerateCreateScript` method and then compare it to the one generated by the `GenerateCreateScript` method.

Reply:
Yeah, I think that makes sense. My comment was more about that we now have 2 separate tests that trigger the same logic, but the only difference is setting a different property on the `DacDeployOptions` class.

Reply:
I've removed the two tests and added one that checks the expected create script

Reply:
Looks good :)